### PR TITLE
Add readout_activation param to models

### LINKED ISSUE #189: using categorical_crossentropy  get loss result is nan
Hi guys,i am a beginner, i have a question here
when i used categorical_crossentropy  , I encountered a "loss : nan" at first epoch, 
I don't know why, can someone help me? thanks 

` model = AttentionSeq2Seq(hidden_dim=hidden_dim, output_length=tar_maxlen,
                                     output_dim=output_dim,input_shape=(input_maxlen, word_vector_size))
  model.compile(loss='categorical_crossentropy', optimizer='adam',metrics=['accuracy'])`

 X = np.zeros((len(inside_data)/2, input_maxlen, word_vector_size), dtype=np.float)#training
 Y = np.zeros((len(inside_data)/2, tar_maxlen,vocab_size),dtype=np.bool)

s_index=0
                        
            for x in range(len(inside_data)/2):

                    if len(inside_data)%2==0:

                        if s_index ==(len(inside_data)/2):
                            break
                        #print dialog_lines_for_train
                        for t_index, token in enumerate(inside_data[s_index]):
                            #print dialog_lines_for_train[s_index][:25]
                            if t_index==tar_maxlen:
                                break
                            X[s_index, t_index] = get_token_vector(token, vector_dict)

                        for t_index, token in enumerate(inside_data[s_index + 1]):
                            if t_index==tar_maxlen:
                                break
                            if token not in word_to_idx:
                                print token
                                continue
                            else :    
                            #print dialog_lines_for_train[s_index+1][:25]
                                Y[s_index,t_index,word_to_idx[token]] = 1
                        s_index=s_index+2

 model.fit(X, Y, batch_size=batch_size, nb_epoch=5,verbose=1)`

the result :
Epoch 1/5
1416/1416 [==============================] - 3s - loss: nan - acc: 0.7864     
Epoch 2/5
1416/1416 [==============================] - 3s - loss: nan - acc: 0.8039     
Epoch 3/5
1416/1416 [==============================] - 3s - loss: nan - acc: 0.8039     
Epoch 4/5
1416/1416 [==============================] - 3s - loss: nan - acc: 0.8039     
Epoch 5/5
1416/1416 [==============================] - 3s - loss: nan - acc: 0.8039

### PR DESCRIPTION
**Pull request by gibrown**
The enables avoiding getting stuck in a NaN loss hole when training. This workaround let's the user fix #189 

Example usage:
```
model = Seq2Seq(input_dim=in_dim, input_length=MAXLENGTH, hidden_dim=HIDDEN_SIZE, output_length=MAXLENGTH, output_dim=out_dim, depth=LAYERS, peek=True, readout_activation='softmax')
```

I do not fully understand the implications of using softmax as the output activation layer, but in my own project (https://github.com/Automattic/wp-translate) setting the output to softmax using this code does seem to have gotten me past getting stuck with NaN during training.

### CODE REVIEW

**Comment by ChristopherLu** [comment]
Does this change include `model.add(TimeDistributed(Dense(output_dim)))`?

**Comment by gibrown** [comment]
@ChristopherLu no, it only let's you change the activation function in the decoder output.

BTW, I have since found that this change did not completely solve my problem. Training worked for longer, but I still eventually ran into NaN losses at some point. I am not sure yet whether this is a problem with the data I am providing the model or if it is some complexity in how the network itself fits together.

**Comment by ChristopherLu** [comment]
@gibrown Exactly. I finally met the NaN problem when the number of training epochs goes up. I guess it's a gradient vanishing problem, and it also depends on the learning rate you set. 

So can we say, the `softmax` activation can only alleviate the NaN, but **not** solve the problem essentially?

**Comment by gibrown** [comment]
@ChristopherLu ya that is my conclusion. Based on this thread and the common problems across applications, I am guessing that it is something inherent in how the functions of the model fit together that for some data you can easily end up in such conditions.

The other possibility is that I have a bug in generating my training data, but I've been looking at that for a while and haven't found it. My next plan (when I get back to this, probably in a few weeks) is to try the Tensorflow seq2seq directly: https://www.tensorflow.org/tutorials/seq2seq

I had tried that method in the past (like a year ago), but been unable to get it to work. I think it has been significantly updated though and that tutorial looks improved. I guess maybe that model could be ported into this lib if that works.

**Comment by ChristopherLu** [comment]
@gibrown Thx. About to try tf seq2seq as well.

**Comment by gibrown** [comment]
@ChristopherLu I've reworked my application to use https://github.com/google/seq2seq and that seems to be working well so far.

**Comment by ChristopherLu** [comment]
@gibrown Thx for the recommendation, I will have a try.

**Comment by cpury** [comment]
This did not make my training converge :(

**Comment by gibrown** [comment]
Ya I don't think this is a real solution to the original problem, I'm just going to close this PR.
